queue.go

```diff
@@ -175,6 +181,9 @@ func (q *DQue) Enqueue(obj interface{}) error {
 		return errors.Wrap(err, "error adding item to the last segment")
 	}
 
+	// Wakeup any goroutine that is currently waiting for an item to be enqueued
+	q.emptyCond.Broadcast()
```

Comment:
Is Broadcast() identical to Signal()?  It seems like it is because there can be only one emptyCond.Wait() happening at a time (given the lock surrounding it).  Just asking.

Reply:
Broadcast wakes up ALL waiting goroutines.
Signal only one of them.

The Wait() call atomically unlocks & sleeps. I added a more descriptive comment to the code.